src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,624 @@
+        // 3 - Both source and destination after leap month in leap year, nothing to do
+        // 4 - Source in leap month, but destination year not leap. Retain month index, preserving day-of-year.
+        // 5 - Source after leap month, but destination year not leap. Move month index back.
+        return resolvePreviousValid (newYear, month, day);
+    }
+
+    /**
+     * Returns a copy of this {@code InternationalFixedDate} with the specified period in months added.
+     * <p>
+     * This method adds the specified amount to the months field in three steps:
+     * <ol>
+     * <li>Add the input months to the month-of-year field</li>
+     * <li>Check if the resulting date would be invalid</li>
+     * <li>Adjust the day-of-month to the last valid day if necessary</li>
+     * </ol>
+     * <p>
+     * For example, 2006-12-13 plus one month would result in the invalid date 2006-13-13.
+     * Instead of returning an invalid result, the last valid day of the month, 2006-13-07, is selected instead.
+     * <p>
+     * This instance is immutable and unaffected by this method call.
+     *
+     * @param monthsToAdd the months to add, may be negative
+     * @return a {@code InternationalFixedDate} based on this date with the months added, not null
+     * @throws DateTimeException if the result exceeds the supported date range
+     */
+    @Override
+    InternationalFixedDate plusMonths (final long monthsToAdd) {
+        if (monthsToAdd == 0) {
+            return this;
+        }
+
+        long calcMonths = Math.addExact (getProlepticMonth (), monthsToAdd);
+        int newYear = ChronoField.YEAR.checkValidIntValue (Math.floorDiv (calcMonths, InternationalFixedChronology.MONTHS_IN_YEAR));
+        int newMonth = Math.toIntExact (calcMonths % InternationalFixedChronology.MONTHS_IN_YEAR) + 1;
+
+        return resolvePreviousValid (newYear, newMonth, getDayOfMonth ());
+    }
+
+    @Override
+    public InternationalFixedDate minus (final TemporalAmount amount) {
+        return (InternationalFixedDate) amount.subtractFrom (this);
+    }
+
+    @Override
+    public InternationalFixedDate minus (final long amountToSubtract, final TemporalUnit unit) {
+        return (amountToSubtract == Long.MIN_VALUE ? plus (Long.MAX_VALUE, unit).plus (1, unit) : plus (-amountToSubtract, unit));
+    }
+
+    //-------------------------------------------------------------------------
+    @Override  // for covariant return type
+    @SuppressWarnings ("unchecked")
+    public ChronoLocalDateTime<InternationalFixedDate> atTime (final LocalTime localTime) {
+        return (ChronoLocalDateTime<InternationalFixedDate>) ChronoLocalDate.super.atTime (localTime);
+    }
+
+    @Override
+    public long until (final Temporal endExclusive, final TemporalUnit unit) {
+        return until (InternationalFixedDate.from (endExclusive), unit);
+    }
+
+    @Override
+    long until (final AbstractDate end, final TemporalUnit unit) {
```

Comment:
`WEEKS` is a pain, given that the leap-day and year-day aren't part of the regular week.  Again, look at how the Discordian Chronology deals with this.